src/lsqecc/pauli_rotations/rotation.py

```diff
@@ -270,9 +270,10 @@ def get_basic_form_decomposition(self) -> List["PauliRotation"]:
             fractions_with_unit_numerator = decompose_pi_fraction(self.rotation_amount)
             output_rotations = []
             for unit_rotation_amount in fractions_with_unit_numerator:
-                new_rotation = copy.deepcopy(self)
-                new_rotation.rotation_amount = unit_rotation_amount
-                output_rotations.append(new_rotation)
+                if unit_rotation_amount.denominator != 1:
```

Comment:
Why is this condition enforced? I am not able to follow

Reply:
We discard rotations by `pi` because they have no effect

Reply:
```suggestion
                # Discard rotations by pi because they have no effect
                if unit_rotation_amount.denominator != 1:
```